src/components/StakingVaults/hooks/useFetchOpportunities.tsx

```diff
@@ -43,51 +29,15 @@ export const useFetchOpportunities = () => {
     { skip: !DynamicLpAssets },
   )
 
-  const queryKey: [
-    string,
-    {
-      requestedAccountIds: AccountId[]
-      portfolioAssetIds: AssetId[]
-      portfolioAccounts: Record<AccountId, PortfolioAccount>
-    },
-  ] = useMemo(() => {
-    return ['fetchAllOpportunities', { requestedAccountIds, portfolioAssetIds, portfolioAccounts }]
-  }, [portfolioAccounts, portfolioAssetIds, requestedAccountIds])
-
-  const { isLoading } = useQuery({
-    queryKey,
-    queryFn: async ({ queryKey: [_, { requestedAccountIds }] }) => {
-      await Promise.all(
-        requestedAccountIds.map(async accountId => {
-          const { chainId } = fromAccountId(accountId)
-          switch (chainId) {
-            case btcChainId:
-            case ltcChainId:
-            case dogeChainId:
-            case bchChainId:
-            case cosmosChainId:
-            case bscChainId:
-            case avalancheChainId:
-              await fetchAllOpportunitiesIdsByChainId(chainId)
-              await fetchAllOpportunitiesMetadataByChainId(chainId)
-              await fetchAllOpportunitiesUserDataByAccountId(accountId)
-              break
-            default:
-              break
-          }
-        }),
-      )
-
-      // We *have* to return a value other than undefined from react-query queries, see
-      // https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#undefined-is-an-illegal-cache-value-for-successful-queries
-      return null
-    },
-    // once the portfolio is loaded, fetch opportunities data
-    enabled: portfolioLoadingStatus !== 'loading' && requestedAccountIds.length > 0,
-    refetchInterval: false,
-    staleTime: Infinity,
-    gcTime: Infinity,
-  })
+  const { isLoading } = useQuery(
+    reactQueries.opportunities.all(
```

Comment:
Fitting the pattern of `react-query` would probably be using `useQueries` here, though I get it that's not necessarily straightforward to make this work with the opportunities abstraction!